src/Engine/MapController.java

```diff
@@ -1,28 +1,34 @@
 package Engine;
 
+import java.util.ArrayList;
+
 import Terrain.Environment;
+import Terrain.GameMap;
 import Terrain.Location;
 import UI.InputHandler;
 import UI.GameMenu;
+import UI.MapView;
 import Units.Unit;
 
 public class MapController
 {
-
   private GameInstance myGame;
+  private MapView myView;
 
-  private enum InputMode {MAP, MOVEMENT, ACTIONMENU, ACTION, PRODUCTION, METAACTION};
+  private enum InputMode {MAP, MOVEMENT, ACTIONMENU, ACTION, PRODUCTION, METAACTION, ANIMATION};
   public enum MetaAction {END_TURN};
 
   private InputMode inputMode;
 
-  // readied Action
-  GameAction currentAction = null;
+  private Path currentMovePath;
 
-  public MapController(GameInstance game)
+  public MapController(GameInstance game, MapView view)
   {
     myGame = game;
+    myView = view;
+    myView.setController(this);
     inputMode = InputMode.MAP;
+    myGame.setCursorLocation(6,5);
```

Comment:
Hah. This wasn't entirely intended. I just did this so I wouldn't have to move to the unit before trying things out.